controllers/vaultstaticsecret_controller.go

```diff
@@ -80,31 +80,32 @@ func (r *VaultStaticSecretReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	}
 
 	var resp *api.KVSecret
+	var respErr error
 	switch o.Spec.Type {
 	case consts.KVSecretTypeV1:
-		w, err := c.KVv1(o.Spec.Mount)
-		if err != nil {
+		if w, err := c.KVv1(o.Spec.Mount); err != nil {
 			return ctrl.Result{}, err
+		} else {
+			resp, respErr = w.Get(ctx, o.Spec.Path)
 		}
-		resp, err = w.Get(ctx, o.Spec.Path)
 	case consts.KVSecretTypeV2:
-		w, err := c.KVv2(o.Spec.Mount)
-		if err != nil {
+		if w, err := c.KVv2(o.Spec.Mount); err != nil {
 			return ctrl.Result{}, err
-		}
-		if o.Spec.Version == 0 {
-			resp, err = w.Get(ctx, o.Spec.Path)
 		} else {
-			resp, err = w.GetVersion(ctx, o.Spec.Path, o.Spec.Version)
+			if o.Spec.Version == 0 {
+				resp, respErr = w.Get(ctx, o.Spec.Path)
+			} else {
+				resp, respErr = w.GetVersion(ctx, o.Spec.Path, o.Spec.Version)
+			}
 		}
 	default:
-		err = fmt.Errorf("unsupported secret type %q", o.Spec.Type)
+		err := fmt.Errorf("unsupported secret type %q", o.Spec.Type)
```

Comment:
I'm pretty sure we cannot hit the `default` code path anyways due to [this check](https://github.com/hashicorp/vault-secrets-operator/blob/3d45188560f240671652109390b1b8d55a6685ac/api/v1beta1/vaultstaticsecret_types.go#L32) in the spec, but good to have anyways I guess, unless I'm mistaken about how kubebuilder works here?

Reply:
True. Good to catch the issue if the marker's value ever gets out of sync with the controller.